sdk/typescript/src/utils/suins.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+const SUI_NS_NAME_REGEX = /^(?:(?:[a-z0-9][a-z0-9-]{0,62})\.?)*@[a-z0-9][a-z0-9-]{0,62}$/i;
+const SUI_NS_DOMAIN_REGEX = /^(?:(?:[a-z0-9][a-z0-9-]{0,62})\.?)+.sui$/i;
+const MAX_SUI_NS_NAME_LENGTH = 200;
+
+export function isValidSuiNSName(name: string): boolean {
+	if (name.length > MAX_SUI_NS_NAME_LENGTH) {
+		return false;
+	}
+
+	if (name.includes('@')) {
+		return SUI_NS_NAME_REGEX.test(name);
+	}
+
+	return SUI_NS_DOMAIN_REGEX.test(name);
+}
+
+export function normalizeSuiNSName(name: string, format: 'at' | 'dot' = 'at'): string {
```

Comment:
My feeling is that we should default to `dot` for now, and move to `at` on the more public release of May.
(and definitely until we integrate these checks in wallets, explorers, etc)

Reply:
Since this is a new function, I'd rather just default to the new format right away (It will have very limited usage before may). We still default to `at` in SuiClient which is the main place people will use this.

Reply:
Ok I read it a few times (got confused because you said `at` 😆 ), I see that on the `resolveNameServiceNames` it has a `dot` default format, so you're right, should be great.